sams-scala-tutorial/serialization-exceptions-and-memory-leaks-no-ws.md

```diff
@@ -13,14 +13,147 @@ Talk:
 
 ## What causes serialization problems
 
-Two things:
+Unserializable data trnasferred across nodes, there are 2 kinds of problems:
 
 1. Spark jobs that *may* require transfer of *object* data between JVMs
     - Trying to broadcast non-serializable data
     - Trying to collect non-serializable data
     - Trying to *shuffle* non-serializable data
 2. Spark jobs that try to transfer non-serializable "*meta*" data between JVMs via (intentional or unintentional) **transative closure capture**.
 
+## Examples for problem 2 and solutions
+```
+object TestSerializable {
+  def main(args: Array[String]): Unit = {
+    val sparkConf = new SparkConf
+    val sc = new SparkContext(sparkConf)
+
+    val data = List(1, 2, 3, 4, 5)
+    val rdd = sc.makeRDD(data)
+
+    /* What is the unserializable task?
+     * When some methods applied in the RDD closure, everything necessary to execute the closure will be packaged up, serialized and sent to executors.
+     * In this process, all functions would be passed into a process called ClosureCleaner. 
+     * If the functions in the closure cannot be cut off from its context, there will be an exception called NotSerializableException.
+     * */
+
+    val unSerializableClassHandler = new NonSerializableClass(1)
+    rdd.map { x => unSerializableClassHandler.func_1 }.count()
+
```

Comment:
@QingyangKong  I do not see a need for this example, surely just calling the method on `NonSerializableObject` directly rather than creating an unnecessary pointer to it will resolve this one.

I.e. `rdd.map(x => NonSerializableObject.func_1(x)).count()`

You have a typo in `NonSerilizableObject`.